tests/puzzles/coin_store.py

```diff
@@ -0,0 +1,91 @@
+from collections import defaultdict
+from dataclasses import dataclass, replace
+from typing import Dict, Iterator, List
+
+from src.consensus.blockchain_check_conditions import blockchain_check_conditions_dict
+from src.types.announcement import Announcement
+from src.types.blockchain_format.coin import Coin
+from src.types.blockchain_format.sized_bytes import bytes32
+from src.types.coin_record import CoinRecord
+from src.types.spend_bundle import SpendBundle
+from src.util.condition_tools import created_announcements_for_conditions_dict, conditions_dict_for_solution
+from src.util.ints import uint32, uint64
+
+
+class BadSpendBundleError(Exception):
+    pass
+
+
+@dataclass
+class CoinTimestamp:
+    seconds: int
+    height: int
+
+
+class CoinStore:
+    def __init__(self):
```

Comment:
Why not use the actual CoinStore?

Reply:
The motivation for these tests was to allow tests on puzzles & solutions that run as fast as possible, so we can quickly iterate on puzzle clvm. I'll look into using the real thing if it's not too inconvenient.

At some point, we want to release a self-contained "contract development kit" ("re-remix") which doesn't need persistence, so a RAM-based CoinStore is sufficient there. I guess I wrote it with this in mind, that these tests would eventually evolve to be that.

Reply:
Ideally, we could get to the place where consensus is factored out of `chia-blockchain` and separated from networking and a particular data store (so any data store that meets the same interface could be plugged in). If we could further separate out "above the line" (blockchain backbone) from "below the line" (transactions), that would be great too, because people debugging contracts don't care about distributed consensus, especially if it slows the development cycle. But it would be really great to use the exact same code as the official client (and just plug in non-distributed consensus & a RAM-based DB).

Reply:
Note that it is already using `blockchain_check_conditions_dict`, which is "below-the-line" consensus, which I think is more valuable than the official `CoinStore`, so we're actually reasonably well on our way to this.

Reply:
OK, i think I see what you're saying. There's very little in the consensus code that actually interacts with clvm. Most of it is in block_body_validation.py